data/src/main/java/org/apache/iceberg/data/orc/GenericOrcWriters.java

```diff
@@ -77,12 +80,12 @@ private GenericOrcWriters() {
     return LongWriter.INSTANCE;
   }
 
-  public static OrcValueWriter<Float> floats() {
-    return FloatWriter.INSTANCE;
+  public static OrcValueWriter<Float> floats(Integer id) {
```

Comment:
The id is required, right? If so, then I think it could be `int` instead. It's an `int` in the constructor that gets called.

This could add a precondition to check that the id is non-null, but I think it would be better to do that before calling this method because the caller would probably know the field name rather than just ID. Using the field name would produce a better error message.